pcsx2-qt/ShortcutCreationDialog.cpp

```diff
@@ -80,400 +77,309 @@ ShortcutCreationDialog::ShortcutCreationDialog(QWidget* parent, const QString& t
-		return;
-	}
+	// Escape provided command line arguments.
+	for (std::string& argument : launch_arguments)
+		ShortcutCreationDialog::EscapeShortcutCommandLineWindows(argument);
 
-	// Shortcut CmdLine Args
-	bool lossless = true;
-	for (std::string& arg : passed_cli_args)
-		lossless &= ShortcutCreationDialog::EscapeShortcutCommandLine(&arg);
+	game_path = Path::ToNativePath(Path::RealPath(game_path));
+	ShortcutCreationDialog::EscapeShortcutCommandLineWindows(game_path);
 
-	if (!lossless)
-		QMessageBox::warning(this, tr("Failed to create shortcut"), tr("File path contains invalid character(s). The resulting shortcut may not work."), QMessageBox::StandardButton::Ok, QMessageBox::StandardButton::Ok);
+	const std::string combined_launch_arguments = fmt::format("{} -- {}", StringUtil::JoinString(launch_arguments.begin(), launch_arguments.end(), " "), game_path);
 
-	ShortcutCreationDialog::EscapeShortcutCommandLine(&clean_path);
-	std::string combined_args = StringUtil::JoinString(passed_cli_args.begin(), passed_cli_args.end(), " ");
-	std::string final_args = fmt::format("{} {} -- {}", combined_args, custom_args, clean_path);
+	const auto str_error = [](HRESULT hresult_error) -> std::string {
+		_com_error err(hresult_error);
+		const TCHAR* error_message = err.ErrorMessage();
+		return fmt::format("{} [{}]", StringUtil::WideStringToUTF8String(error_message), hresult_error);
+	};
 
-	Console.WriteLnFmt("Creating a shortcut '{}' with arguments '{}'", link_file, final_args);
+	HRESULT hresult = CoInitialize(NULL);
+	if (FAILED(hresult))
+	{
+		ShortcutCreationDialog::ReportShortcutCreationFailure(tr("CoInitialize failed (%1)").arg(str_error(hresult)));
+		return;
+	}
 
-	const auto str_error = [](HRESULT hr) -> std::string {
-		_com_error err(hr);
-		const TCHAR* errMsg = err.ErrorMessage();
-		return fmt::format("{} [{}]", StringUtil::WideStringToUTF8String(errMsg), hr);
-	};
+	// Create file path for shortcut.
+	PWSTR base_directory = nullptr;
+	SHGetKnownFolderPath(is_desktop ? FOLDERID_Profile : FOLDERID_StartMenu, 0, NULL, &base_directory);
 
-	// Construct the shortcut
-	// https://stackoverflow.com/questions/3906974/how-to-programmatically-create-a-shortcut-using-win32
-	HRESULT res = CoInitialize(NULL);
-	if (FAILED(res))
+	if (FAILED(hresult))
 	{
-		Console.ErrorFmt("Failed to create shortcut: CoInitialize failed ({})", str_error(res));
-		QMessageBox::critical(this, tr("Failed to create shortcut"), tr("CoInitialize failed (%1").arg(str_error(res)), QMessageBox::StandardButton::Ok, QMessageBox::StandardButton::Ok);
+		ShortcutCreationDialog::ReportShortcutCreationFailure(tr("Failed to get path to %1 (%2)")
+																.arg(is_desktop ? "desktop" : "start menu", str_error(hresult)));
 		return;
 	}
 
-	Microsoft::WRL::ComPtr<IShellLink> pShellLink;
-	Microsoft::WRL::ComPtr<IPersistFile> pPersistFile;
+	std::filesystem::path shortcut_file_path(base_directory);
```

Comment:
PCSX2 currently doesn't use std::filesystem anywhere else. I'm not opposed to switching to it personally, but some other people on the team should probably comment.